ormar/models/metaclass.py

```diff
@@ -195,6 +197,33 @@ def verify_constraint_names(
             )
 
 
+def get_constraint_copy(
+    constraint: ColumnCollectionConstraint,
+) -> Union[UniqueColumns, IndexColumns, CheckColumns]:
+    """
+    Copy the constraint and unpacking it's values
+
+    :raises ValueError: if non subclass of ColumnCollectionConstraint
+    :param value: an instance of the ColumnCollectionConstraint class
+    :type value: Instance of ColumnCollectionConstraint child
+    :return: copy ColumnCollectionConstraint ormar constraints
+    :rtype: Union[UniqueColumns, IndexColumns, CheckColumns]
+    """
+
+    constraints = {
+        sqlalchemy.UniqueConstraint: lambda x: UniqueColumns(*x._pending_colargs),
+        sqlalchemy.Index: lambda x: IndexColumns(*x._pending_colargs),
+        sqlalchemy.CheckConstraint: lambda x: CheckColumns(x.sqltext),
+    }
+    checks = (key if isinstance(constraint, key) else None for key in constraints)
+    target_class = next((target for target in checks if target is not None), None)
+    constructor: Optional[Callable] = constraints.get(target_class)
+    if not constructor:
+        raise ValueError(f"{constraint} must be a ColumnCollectionMixin!")
```

Comment:
If this is the case why not use `issubclass(target_class, ColumnCollectionMixin)`?

Reply:
As you see we need a different attribute of the "parent" (either `_pending_colargs` or `sqltext`). 

You have to unpack the constraints attributes otherwise it will not work with inheritance as the constraints would have been bound to the parent model and fail. All of those (unique, index and check) constraints are subclasses od `ColumnCollectionMixin` so it's not enough as a check.